tests/PactNet.Native.Tests/NativeRequestBuilderTests.cs

```diff
@@ -93,6 +93,14 @@ public void WithQuery_RepeatedQuery_SetsIndex()
             this.mockServer.Verify(s => s.WithQueryParameter(this.handle, "other", "value", 0));
         }
 
+        [Fact]
+        public void WithHeader_WhenCalledWithMatcher_AddsSerialisedHeaderParam()
+        {
+            this.builder.WithHeader("name", PactNet.Matchers.Match.Regex("header", "^header$"));
+
+            this.mockServer.Verify(s => s.WithRequestHeader(this.handle, "name", "{\"pact:matcher:type\":\"regex\",\"value\":\"header\",\"regex\":\"^header$\"}", 0));
+        }
+
```

Comment:
Needs a test which makes sure it respects the serialisation settings, although given we only support matchers is that necessary? I'm not sure you could customise the serialisation settings for those and still have them compatible with the Rust core.

Reply:
I remove the option to customise this